content/platforms/kubernetes/tasks/ldap-on-k8s.md

```diff
@@ -0,0 +1,501 @@
+
+```
+kubectl port-forward pod/ldap-0 9443
+```
+
+Then you change the LDAP configuration with a JSON version of the ldap configuration.
+
+```
+cat << EOF > tmp.json
+{"saslauthd_ldap_conf": "ldap_servers: ldap://ldap-openldap.bdb.svc:389\nldap_search_base: ou=users,dc=example,dc=org\nldap_filter: (uid=%u)\nldap_bind_dn: cn=admin,dc=example,dc=org\nldap_password: admin\n"}
+EOF
+curl -f -k -u "demo@redislabs.com:xxx" -X PUT -d @tmp.json -H "Content-Type: application/json" https://localhost:9443/v1/cluster
+```
+
+The key value for “saslauthd_ldap_conf” is the configuration file verbatim as it would be in the saslauthd configuration file. As such, it can be relatively tricky to get the syntax right and generating this programmatically is helpful. A little python program (make_request.py) can help:
+
+```
+import sys
+import json
+
+with open(sys.argv[2]) as data:
+   d = {}
+   d[sys.argv[1]] = data.read()
+   print(json.dumps(d))
+```
+
+And so we can do:
+
+```
+python make_request.py saslauthd_ldap_conf ldap.conf > tmp.json
+curl -f -k -u "demo@redislabs.com:xxx" -X PUT -d @tmp.json -H "Content-Type: application/json" https://localhost:9443/v1/cluster
+```
+
+### Automating LDAP Configuration
+
+If you are automating deployments with the operator, you can automate this configuration change too with a simple K8s job.
+
+1. Create a ConfigMap that contains the LDAP configuration:
+
+   ```
+   kubectl create configmap config-job --from-file=saslauthd_ldap_conf=ldap.conf
+   ```
+
+2. Submit this job:
+
+   ```
+   apiVersion: batch/v1
+   kind: Job
+   metadata:
+     name: config-ldap
+   spec:
+     backoffLimit: 4
+     template:
+       spec:
+         serviceAccountName: redis-enterprise-operator
+         restartPolicy: Never
+         containers:
+           - name: config
+             image: you/job-config-ldap:1
+             env:
+               - name: MY_NAMESPACE
+                 valueFrom:
+                   fieldRef:
+                     fieldPath: metadata.namespace
+               - name: CLUSTER_NAME
+                 value: ldap
+               - name: CLUSTER_USER
+                 valueFrom:
+                  secretKeyRef:
+                    name: ldap
+                    key: username
+               - name: CLUSTER_PASSWORD
+                 valueFrom:
+                  secretKeyRef:
+                    name: ldap
+                    key: password
+             volumeMounts:
+               - name: config
+                 mountPath: "/config"
+                 readOnly: true
+         volumes:
+           - name: config
+             configMap:
+               name: config-job
+   ```
+
+   In the above job, the cluster name is “ldap”. You will have to change this to the name of your cluster in all three of the CLUSTER_NAME, CLUSTER_USER, and CLUSTER_PASSWORD environment variables.
```

Comment:
Can you choose a more realistic cluster name here? It may confuse the user to call the cluster "ladp"